libs/block-template-react/package.json

```diff
@@ -19,13 +19,13 @@
   },
   "scripts": {
     "build": "block-scripts build",
+    "codegen": "rimraf ./src/types && block-scripts codegen && yarn format && yarn fix:eslint",
```

Comment:
Calling format and fix is causing problems in our CI 🙃 
https://github.com/blockprotocol/blockprotocol/actions/runs/4406199711/jobs/7718107593#step:6:198

@CiaranMn should I just remove the eslint call, or should I try investing time into our bespoke package chore thing to account for this?

Reply:
Just remove it. I'd also scope the `format` call to only the autogenerated file(s), otherwise running `codegen` is going to cause a diff in other files too, which seems unexpected.

Reply:
Also seems like it might be bad to remove `src/types` if the user has added other files there? can we filter all these to extension (if we still have `.gen.ts`)?

Reply:
> I'd also scope the format call to only the autogenerated file(s),
> Also seems like it might be bad to remove src/types if the user has added other files there? can we filter all these to extension (if we still have .gen.ts)?

I thought about scoping these, but the users are responsible for picking the file names and root folder now (in the config below) so it would be difficult to automate..

Reply:
You could output the files affected from the `codegen` script so it can be piped into the `format`? or just put the `format` inside the codegen script

Reply:
or `format` could read from the codegen config (but returning the generated files might generally be useful)

Reply:
[Output affected files on stdout and use as input for formatting](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/98cf8fa0230049e81e049b5b9f78caa386cf8f3e)

Reply:
[Clean outputDir from within codegen](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/2c764b7c21dab05f2c3749dd85df0a4d3cb10195)

Reply:
Formatting looks good. I'm still wary of the combination of (a) nuking the `outputFolder` and (b) having it default in the template to something as generic as `src/types`. What about making the default folder something more specific, e.g. `src/types/generated` or something?  Then people can choose to simplify it if they want. The current set up prevents people from using `src/types` without changing the output.

Reply:
Also wonder if we should call the default file `block.ts` or `block-entity.ts`? otherwise initial set up for a block (or swapping the block entity) means immediately replacing both the filename and the URL

Reply:
I've changed it to `src/types/generated` as you suggest. I've left it as `thing.ts` because I think it's good to get someone to replace the filename and get into the thought of separating types into well-named files as they need to. `block-entity.ts` feels so generic and isn't a pattern I'd want to encourage personally. I don't feel super strongly so I can rename it if you do
[Change default codegen outputFolder to src/types/generated](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/498eac132cf838737726429d49b0bb53f84c0961)

Reply:
I ended up changing my mind about this while writing the docs, as I realised not setting the `schema` explicitly was quite strange and we were depending on it being in a somewhat unrelated thing of `codegen`.

I've renamed `schema` to `blockEntityType` within packages which use the `package.json` approach, and updated the `codegen` to allow that to be injected:
[Replace schema with blockEntityType and inject that into codegen …](https://github.com/blockprotocol/blockprotocol/pull/1158/commits/d8129045ce1ba4622c0b3c3d3cb191ec276698be)